.circleci/main.yml

```diff
@@ -87,6 +87,112 @@ jobs:
+      - run: go install gotest.tools/gotestsum@latest
+      - restore_cache:
+          keys:
+            - pach-build-dependencies-v2-{{ checksum "etc/testing/circle/install.sh" }}
+      - run: etc/testing/circle/install.sh
+      - go/install:
+          version: << pipeline.parameters.go-version >>
+      - run:
+          name: Start minikube
+          command: etc/testing/circle/start-minikube.sh
+          background: true
+      # The build cache will grow indefinitely, so we rotate the cache once a week.
+      # This ensures the time to restore the cache isn't longer than the speedup in compilation.
+      - run: "echo $(($(date +%s)/604800)) > current_week"
+      - restore_cache:
+          keys:
+            - pach-go-build-cache-v1-{{ .Branch }}-{{ checksum "current_week" }}
+            - pach-go-build-cache-v1-master-{{ checksum "current_week" }}
+
+      # Only restore the module cache based on an exact match for go.sum.
+      # This also avoids accumulating old versions of modules over time.
+      - restore_cache:
+          keys:
+            - pach-go-mod-cache-v2-{{ checksum "go.sum" }}
+      # Try running go mod download to pull the rest of the deps
+      - run: go mod download
+      - run: etc/testing/circle/wait-minikube.sh
+      - run:
+          name: Install minio
+          command: kubectl apply -f etc/testing/minio.yaml
+      - run:
+          name: Install prom # Make Launch stats
+          command: kubectl apply --filename etc/kubernetes-prometheus -R
+      - run:
+          name: Wait for docker images to be built
+          command: etc/testing/circle/wait_for_docker_images.sh
+      - run:
+          name: run tests
+          no_output_timeout: 20m
+          command: |
+            PROM_PORT=$(kubectl --namespace=monitoring get svc/prometheus -o json | jq -r .spec.ports[0].nodePort)
+            export PROM_PORT
+
+            bucket_num=$CIRCLE_NODE_INDEX
+            num_buckets=$CIRCLE_NODE_TOTAL
+            package="./src/server"
+
+            tests=( $(go test -v -tags=k8s  "${package}" -list ".*" | grep -v '^ok' | grep -v '^Benchmark') )
+            # Add anchors for the regex so we don't run collateral tests
+            tests=( "${tests[@]/#/^}" )
+            tests=( "${tests[@]/%/\$\$}" )
+            total_tests="${#tests[@]}"
+
+            # Determine the offset and length of the sub-array of tests we want to run
+            # The last bucket may have a few extra tests, to accommodate rounding
+            # errors from bucketing:
+            let "bucket_size=total_tests/num_buckets" \
+              "start=bucket_size * (bucket_num-1)" \
+              "bucket_size+=bucket_num < num_buckets ? 0 : total_tests%num_buckets"
+
+            test_regex="$(IFS=\|; echo "${tests[*]:start:bucket_size}")"
+
+            echo $test_regex
+
+            gotestsum \
+              --junitfile ${TEST_RESULTS}/gotestsum-report.xml \
+              --rerun-fails \
```

Comment:
Maximum rerun attempts defaults to 2, and can be changed with --rerun-fails=n good enough?

Reply:
Are we good with the default 2 for now?